smoketest/scripts/cli/test_load_balancing_inbound-ipvsadm.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2020 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import jmespath
+import json
+import unittest
+
+from base_vyostest_shim import VyOSUnitTestSHIM
+
+from vyos.configsession import ConfigSessionError
+from vyos.util import read_file
+from vyos.template import is_ipv6
+
+base_path = ['load-balancing', 'inbound', 'ipvsadm']
+virtual_ip = '127.0.0.1'
+virtual_ip6 = 'fc00::1'
+IPVSADM_CONF = '/tmp/ipvsadm.rules'
+
+class TestLoadBalancingInbound(VyOSUnitTestSHIM.TestCase):
+    def setUp(self):
+        # ensure we can also run this test on a live system - so lets clean
+        # out the current configuration :)
+        self.cli_delete(base_path)
+
+    def tearDown(self):
+        self.cli_delete(base_path)
+        self.cli_commit()
+    
+    def test_load_balancing_inbound_duplicate_virtual_service_rule(self):
+        rule = 0
+        backend_port = 18032
+        virtual_port = 8080
+        backend_id = 0
+        for protocol in ['tcp', 'udp', 'sctp']:
+            for mode in ['direct', 'tunnel', 'nat']:
+                for ip in [virtual_ip, virtual_ip6]:
+                    rule += 1
+                    backend_port +=1 
+                    backend_id += 1
+                    self.cli_set(base_path + ['rule', str(rule), 'virtual-address', ip])
+                    self.cli_set(base_path + ['rule', str(rule), 'port', str(virtual_port)])
+                    self.cli_set(base_path + ['rule', str(rule), 'mode', mode])
+                    self.cli_set(base_path + ['rule', str(rule), 'protocol', protocol])
+        
+        with self.assertRaises(ConfigSessionError):
```

Comment:
Please add a comment why an exception is raised here.

Reply:
done